pyproject2setuppy/setuptools.py

```diff
@@ -0,0 +1,34 @@
+# pyproject2setup.py -- setuptools support
+# vim:se fileencoding=utf-8 :
+# (c) 2020 Eli Schwartz
+# 2-clause BSD license
+
+import os
+import tokenize
+
+def handle_setuptools(data):
+    """
+    Handle pyproject.toml unserialized into data, by ignoring it and using the
+    setuptools build system instead.
+
+    Prefer running the contents of setup.py, but fall back to running a setup()
+    function.
+    """
+    if os.path.exists('setup.py'):
+        with getattr(tokenize, 'open', open)('setup.py') as f:
```

Comment:
Any reason not to just `os.execv()` into `sys.executable` with the argument?

Though I'm not sure if this case really needs handling, to be honest. In our use case, the program is injected as `setup.py` only if the file originally does not exist.

Reply:
Two reasons:
- forking is mildly wasteful, though granted hardly noticeable
- it's possible for setup.py to import distutils but not setuptools, and pip uses the same approach of importing setuptools to monkey patch distutils, then exec'ing the contents of the file to ensure distutils is always used. I guess a reasonable response is "no one who is using a pyproject.toml is ever importing distutils, and we are not pip so don't care about supporting or fixing arbitrary setup.py".

I do believe it should still be handled, though, because you wouldn't want to badly misbehave if setup.py does exist and someone used this module despite not being strictly necessary.

I guess os.execv isn't a big deal in that case.